frontend/src/components/MapPanel.tsx

```diff
@@ -75,6 +84,23 @@ function MapPanel({
     const box = extractBoundingBox();
     if (box.length > 0) {
       onBoundingBoxChange(extractBoundingBox());
+      (searchBox as any).current = getSearchedAreaBox();
+    }
+  };
+
+  const updateAnnotationMode = (updatedMode: string) => {
+    console.log(updatedMode);
+    setAnnotationMode(updatedMode);
+
+    switch (annotationMode) {
```

Comment:
ur click twice bug is probably bc you are calling the switch for state annotationMode here.

`setAnnotationMode` is async so it will not have updated by the time the switch fires.

The correct way to do this is to put the switch in a `useEffect` and subscribe to `annotationMode` state changes.

https://reactjs.org/docs/hooks-effect.html .

This might not be the only reason the bug is appearing but certainly moving switch into `useEffect` will make life better.